src/common/config_parser/config_parser.cpp

```diff
@@ -227,26 +228,30 @@ ExpectedBool MenderConfigFromFile::LoadFile(const string &path) {
 				}
 			}
 		}
+		artifact_verify_keys_field_used_ = true;
 	}
 
 	e_cfg_value = cfg_json.Get("ArtifactVerifyKey");
 	if (e_cfg_value) {
 		const json::Json value_json = e_cfg_value.value();
 		const json::ExpectedString e_cfg_string = value_json.GetString();
 		if (e_cfg_string) {
-			if (artifact_verify_keys.size() != 0) {
+			if (artifact_verify_keys_field_used_) {
 				auto err = MakeError(
 					ConfigParserErrorCode::ValidationError,
 					"Both 'ArtifactVerifyKey' and 'ArtifactVerifyKeys' are set");
 				return expected::unexpected(err);
 			}
+			assert(artifact_verify_keys.size() <= 1);
```

Comment:
Q: Why?

Reply:
Because if `artifact_verify_keys_field_used_` is false, then it should not be possible to have more than one key in the list, since only the `ArtifactVerifyKey` singleton form has been used. We're just making sure of that logic here.

Reply:
But why did you remove the test for the multiple elements case? :thinking:

Reply:
To be accurate it was a test for *any* element, not multiple elements. And we cannot use that test anymore because we need to be able to replace an element with a new element (the override case, first bullet point in the commit message). So there can be an element already, or no element. We clear the list, so it makes no difference in the lines below.

There should never be more than one element, because then `artifact_verify_keys_field_used_` will be true (the only way to get multiple elements into the list).

Reply:
Roger :green_circle: